hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/kafka/KafkaFacade.java

```diff
@@ -109,62 +103,59 @@
   @EJB
   Settings settings;
   @EJB
-  private ProjectFacade projectsFacade;
-  @EJB
-  private CertificateMaterializer certificateMaterializer;
-  @EJB
   private BaseHadoopClientsService baseHadoopService;
+  //TODO: added temporarly to compile the unrefactored endpoints - remove
   @EJB
-  private HdfsUsersController hdfsUsersController;
-  @EJB
-  private UserFacade userFacade;
-
-  private static final String COLON_SEPARATOR = ":";
-  public static final String SLASH_SEPARATOR = "//";
-  public static final String KAFKA_SECURITY_PROTOCOL = "SSL";
-  private static final String KAFKA_BROKER_EXTERNAL_PROTOCOL = "EXTERNAL";
-  private static final String PROJECT_DELIMITER = "__";
-  public static final String DLIMITER = "[\"]";
-
+  private KafkaController kafkaController;
+  
+  private AdminClient adminClient;
+  
   protected EntityManager getEntityManager() {
     return em;
   }
 
   public KafkaFacade() {
   }
-
-  /**
-   * Get all the Topics for the given project.
-   *
-   * @param project
-   * @return
-   */
-  public List<TopicDTO> findTopicsByProject(Project project) {
-
-    List<ProjectTopics> res = em.createNamedQuery("ProjectTopics.findByProject", ProjectTopics.class)
-        .setParameter("project", project)
-        .getResultList();
-
-    List<TopicDTO> topics = new ArrayList<>();
-    if(res != null && !res.isEmpty()) {
-      topics = new ArrayList<>();
-      for (ProjectTopics pt : res) {
-        topics.add(new TopicDTO(pt.getTopicName(),
-            pt.getSchemaTopics().getSchemaTopicsPK().getName(),
-            pt.getSchemaTopics().getSchemaTopicsPK().getVersion()));
-      }
-    }
-    return topics;
+  
+  @PostConstruct
+  private void init() {
+    adminClient = getAdminClient();
```

Comment:
@kai-chi It might be better to move this adminClient to its own singleton. The pool will contain many KafkaFacade instances, do we need one adminClient for each instance?
@ErmiasG any opinion?

Reply:
introduced a singleton :+1:

Reply:
with singletons we need to be careful in case the client times out, although it is instantiated. Subsequent calls to the client would then fail. Is there an API to get the status of the client?

Reply:
there's no API for status

Reply:
it seems like it's taking care of timeouts. in [1] it sets retries and reconnects settings

[1] https://github.com/apache/kafka/blob/ac385c4c3a770728848438f28f4acb8854ffc868/clients/src/main/java/org/apache/kafka/clients/admin/AdminClientConfig.java

Reply:
successfully created topics after the AdminClient times out (the default is 5 min), meaning it recreates after time out.